sentry-core/src/scope/real.rs

```diff
@@ -204,7 +204,54 @@ impl Scope {
         Arc::make_mut(&mut self.contexts).remove(key);
     }
 
+    /// Returns a mutable reference to a context key if it exists.
+    ///
+    /// This allows in-place modification of a context value, especially useful when dealing
+    /// with [`Context::Other`] and need to do something more complicated then the
+    /// [`Scope::set_context_other`] shortcut allows.
+    ///
+    /// If the `key` does not exist `None` is returned.
+    pub fn context_mut(&mut self, key: &str) -> Option<&mut BTreeMap<String, Value>> {
```

Comment:
We don’t really want to expose mutable refs into the scope itself, as that would in this specific case expose the inner `BTreeMap`, which we just recently changed from (it used to be a map from the unmaintained `im` crate).

Also, provided you could always return `None`, the `noop.rs` build should actually noop all these methods.

Reply:
err, the `BTreeMap` is entirely part of the public API currently: https://docs.rs/sentry/0.22.0/sentry/protocol/enum.Context.html#variant.Other

Reply:
Oh, my bad, I was confusing which map you were actually returning. Though still not convinced we want to have this API 🤔

Reply:
I'm not convinced of the naming, but currently you don't have enough api to add multiple entries to the same context key in different `configure_scope()` calls.  I think @jan-auer also complain about something unified API with this addition.  But I still think you can't do something very reasonable with the current API.